canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/CanaryListFragmentWorker.java

```diff
@@ -0,0 +1,111 @@
+                .withCredentials(credentialsProvider)
+                .withEndpointConfiguration(new AwsClientBuilder.EndpointConfiguration(endPoint, region.getName()))
+                .build();
+    }
+
+    public static CanaryListFragmentWorker create(final String streamName,
+                                                  final AWSCredentialsProvider credentialsProvider,
+                                                  final Regions region,
+                                                  final AmazonKinesisVideo amazonKinesisVideo,
+                                                  final FragmentSelector fragmentSelector) {
+        final GetDataEndpointRequest request = new GetDataEndpointRequest()
+                .withAPIName(APIName.LIST_FRAGMENTS).withStreamName(streamName);
+        final String endpoint = amazonKinesisVideo.getDataEndpoint(request).getDataEndpoint();
+        
+        return new CanaryListFragmentWorker(
+                streamName, credentialsProvider, endpoint, region, fragmentSelector);
+    }
+
+    private void close() {
+        this.mAmazonKinesisVideoArchivedMedia.shutdown();
+    }
+
+    @Override
+    public List<CanaryFragment> call() {
+        List<CanaryFragment> fragments = new ArrayList<>();
+        try {
+            log.info("Start CanaryListFragment worker on stream {}", this.mStreamName);
+            
+            ListFragmentsRequest request = new ListFragmentsRequest()
+                    .withStreamName(this.mStreamName).withFragmentSelector(this.mFragmentSelector).withMaxResults(this.mFragmentsPerRequest);
+
+            ListFragmentsResult result = this.mAmazonKinesisVideoArchivedMedia.listFragments(request);
+            
+            log.info("List Fragments called on stream {} response {} request ID {}",
+                    this.mStreamName,
+                    result.getSdkHttpMetadata().getHttpStatusCode(),
+                    result.getSdkResponseMetadata().getRequestId());
+
+            for (Fragment f : result.getFragments()) {
+                fragments.add(new CanaryFragment(f));
+            }
+            String nextToken = result.getNextToken();
+
+            /* If result is truncated, keep making requests until nextToken is empty */
+            while (nextToken != null) {
+                request = new ListFragmentsRequest()
+                        .withStreamName(this.mStreamName).withNextToken(nextToken);
+                result = this.mAmazonKinesisVideoArchivedMedia.listFragments(request);
+
+                for (Fragment f : result.getFragments()) {
+                    fragments.add(new CanaryFragment(f));
+                }
+                nextToken = result.getNextToken();
+            }
+
+            // Fragments in the return of a listFragments call are in no particular order,
+            // so let's sort them by ascending fragment number
+            fragments.sort(Comparator.comparing(CanaryFragment::getFragmentNumberInt));
```

Comment:
Wondering if the sorting is necessary. What is it used for?

Reply:
Not currently used for anything, but could be used to validate fragment order as media is ingested and to identify specific missing fragments in continuity. Will re-evaluate the necessity.